runtime/ftplugin/vim.vim

```diff
@@ -51,9 +52,9 @@ setlocal keywordprg=:help
 
 " Comments starts with # in Vim9 script.  We have to guess which one to use.
 if "\n" .. getline(1, 32)->join("\n") =~# '\n\s*vim9\%[script]\>'
-  setlocal commentstring=#%s
+  setlocal commentstring=#\ %s
 else
-  setlocal commentstring=\"%s
+  setlocal commentstring=\"\ %s
```

Comment:
Using a space here may not always work, as mentioned in `:h line-continuation-comment`.